pkg/apis/sessions/session_state_test.go

```diff
@@ -148,98 +29,132 @@ func TestExpired(t *testing.T) {
+			IDToken:           "IDToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			CreatedAt:         &created,
+			ExpiresOn:         &expires,
+			RefreshToken:      "RefreshToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
 		},
-		{
-			SessionState: SessionState{
-				Email:        "user@domain.com",
-				User:         "just-user",
-				AccessToken:  "token1234",
-				IDToken:      "rawtoken1234",
-				CreatedAt:    &c,
-				ExpiresOn:    &e,
-				RefreshToken: "refresh4321",
-			},
-			Encoded: `{"Email":"user@domain.com","User":"just-user"}`,
+		"No ExpiresOn": {
+			Email:             "username@example.com",
+			User:              "username",
+			PreferredUsername: "preferred.username",
+			AccessToken:       "AccessToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			IDToken:           "IDToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			CreatedAt:         &created,
+			RefreshToken:      "RefreshToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+		},
+		"No PreferredUsername": {
+			Email:        "username@example.com",
+			User:         "username",
+			AccessToken:  "AccessToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			IDToken:      "IDToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			CreatedAt:    &created,
+			ExpiresOn:    &expires,
+			RefreshToken: "RefreshToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+		},
+		"Minimal session": {
+			User:         "username",
+			IDToken:      "IDToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			CreatedAt:    &created,
+			RefreshToken: "RefreshToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+		},
+		"Bearer authorization header created session": {
+			Email:       "username",
+			User:        "username",
+			AccessToken: "IDToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			IDToken:     "IDToken.12349871293847fdsaihf9238h4f91h8fr.1349f831y98fd7",
+			ExpiresOn:   &expires,
 		},
 	}
 
-	for i, tc := range testCases {
-		encoded, err := tc.EncodeSessionState(tc.Cipher)
-		t.Logf("i:%d Encoded:%#vSessionState:%#v Error:%#v", i, encoded, tc.SessionState, err)
-		if tc.Error {
-			assert.Error(t, err)
-			assert.Empty(t, encoded)
-			continue
-		}
-		assert.NoError(t, err)
-		assert.JSONEq(t, tc.Encoded, encoded)
+	for name, ss := range testCases {
+		t.Run(name, func(t *testing.T) {
+			encoded, err := ss.EncodeSessionState(c, false)
+			assert.NoError(t, err)
+			encodedCompressed, err := ss.EncodeSessionState(c, true)
+			assert.NoError(t, err)
+			assert.GreaterOrEqual(t, len(encoded), len(encodedCompressed))
+
+			decoded, err := DecodeSessionState(encoded, c, false)
+			assert.NoError(t, err)
+			decodedCompressed, err := DecodeSessionState(encodedCompressed, c, true)
+			assert.NoError(t, err)
+
+			compareSessionStates(t, decoded, decodedCompressed)
+			compareSessionStates(t, decoded, &ss)
+		})
```

Comment:
Is it worth possibly checking that trying to decode with a different cipher results in an error?

Reply:
I think we might be sufficient covered already by these tests in the cipher section, what do you think?

https://github.com/oauth2-proxy/oauth2-proxy/blob/c4cf15f3e1aaa8f8ae700e1c844f08380421315b/pkg/encryption/cipher_test.go#L161

Reply:
I guess the middle encryption layer in these tests is sorta passed over and we never explicitly test changing the cipher breaks the functions (vs our compression size & different output checks in the middle stage).

I'll give it a whirl adding it.

Reply:
I ended up doing this -- and took the opportunity to test with all valid AES secret lengths and ran the tests through all our potential ciphers. Before I was using the const secret & just the AES-CFB cipher.